### PR TITLE
[SPARK-46727][SQL] Port `classifyException()` in JDBC dialects on error classes

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to port the existing `classifyException()` method which accepts a description to new one w/ an error class added by https://github.com/apache/spark/pull/44358. The modified JDBC dialects are: DB2, H2, Oracle, MS SQL Server, MySQL and PostgreSQL.

### Why are the changes needed?
The old method `classifyException()` which accepts a `description` only has been deprecated already by ...

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
By existing integration tests, and the modified test suite:
```
$ build/sbt "test:testOnly *JDBCV2Suite"
```

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by MaxGekk** [comment]
cc @milastdbx

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/main/scala/org/apache/spark/sql/jdbc/H2Dialect.scala [hunk 1]
```diff
@@ -224,25 +228,21 @@ private[sql] object H2Dialect extends JdbcDialect {
             throw new NoSuchNamespaceException(errorClass = "SCHEMA_NOT_FOUND",
               messageParameters = Map("schemaName" -> quotedName))
           // INDEX_ALREADY_EXISTS_1
-          case 42111 =>
-            // The message is: Failed to create index indexName in tableName
-            val regex = "(?s)Failed to create index (.*) in (.*)".r
-            val indexName = regex.findFirstMatchIn(message).get.group(1)
-            val tableName = regex.findFirstMatchIn(message).get.group(2)
+          case 42111 if errorClass == "FAILED_JDBC.CREATE_INDEX" =>
```
Comment (beliefer) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/H2Dialect.scala:231]: Is there only need error code?

**Review by beliefer — APPROVED** [review]
LGTM except one comment.

File: sql/core/src/main/scala/org/apache/spark/sql/jdbc/H2Dialect.scala (hunk 1 above)
Comment (MaxGekk) [reply sql/core/src/main/scala/org/apache/spark/sql/jdbc/H2Dialect.scala:231]: It is some kind of a guard for accessing `messageParameters`. Previous implementation implicitly assumes that the error happens on creating an index, see the regexp. Here I just made the assumption explicit.

Comment (beliefer) [reply sql/core/src/main/scala/org/apache/spark/sql/jdbc/H2Dialect.scala:231]: SGTM.

**Comment by MaxGekk** [comment]
Merging to master. Thank you, @beliefer and @cloud-fan for review.

**Pull request closed without merging**